bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/util/StateUtil.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mikrotik.internal.util;
+
+import java.math.BigInteger;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * The {@link StateUtil} class holds static methods to cast Java native/class types to OpenHAB values
+ *
+ * @author Oleg Vivtash - Initial contribution
+ */
+@NonNullByDefault
+public class StateUtil {
+
+    public static State stringOrNull(@Nullable String value) {
+        return value == null ? UnDefType.NULL : new StringType(value);
+    }
+
+    public static State intOrNull(@Nullable Integer value) {
+        return value == null ? UnDefType.NULL : new DecimalType(value.floatValue());
+    }
+
+    public static State bigIntOrNull(@Nullable BigInteger value) {
+        return value == null ? UnDefType.NULL : DecimalType.valueOf(value.toString());
+    }
+
+    public static State floatOrNull(@Nullable Float value) {
+        return value == null ? UnDefType.NULL : new DecimalType(value);
```

Comment:
If you use Units of Measure like `Number:DataAmount`, the State needs to be of type `QuantityType`.

Reply:
Done